test/Test/Iris/IO.hs

```diff
@@ -0,0 +1,34 @@
+module Test.Iris.IO (ioSpec)
+where
+
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+-- Silently has side effect: writes file to tmp (or current dir) and then deletes it.
+
+import System.IO (stderr, stdout)
+import System.IO.Silently (hCapture_, hSilence)
+
+import qualified Iris.IO as IO (err, errLn, out, outLn)
+
+checkStdErr :: IO a -> IO String
+checkStdErr = hCapture_ [stderr] . hSilence [stdout]
```

Comment:
I think the `hSilence` part should be removed:

```suggestion
checkStdErr = hCapture_ [stderr]
```

Otherwise, the test `it "does not write to stderr " $ checkStdErr (IO.out "TEXT") >>= flip shouldBe ""` doesn't really check that the function doesn't output anything to `stderr`. Because `stderr` was explicitly silenced.

Reply:
we need to silence the handle we're not checking. otherwise erroneous behaviour will cause "TEXT" to leak into the console, messing up the test progress report. 

I believe the implementation **is** correct as it is. checkStdErr does not silence stderr. it silences stdout. 

i can also confirm that the tests **do** seem to work; hacking IO.out to print to both stdout and stderr gives ths following result:
```
   IO
    out
      writes to stdout, no LF  [✔]
      does not write to stderr  [✘]
    outLn
      writes to stdout, LF  [✔]
      does not write to stderr  [✔]
    err
      writes to sterr, no LF  [✔]
      does not write to stdout  [✔]
    errLn
      writes to stderr, LF  [✔]
      does not write to stdout  [✔]

Failures:

  test/Test/Iris/IO.hs:25:83:
  1) Iris.IO.out does not write to stderr
       expected: ""
        but got: "TEXT"
```

but maybe i'm still missing something...

Reply:
I see now. I was confused by `silence` but I can see that both functions `checkStdOut` and `checkStdErr` are used in a pair to test the output to both `stdout` and `stderr`. A single test is not enough to make sure that the function doesn't print somewhere else, but two tests together make everything work perfectly 👍🏻 

No need to change these helper functions 😌